File: apps/nihongo_flutter/lib/data/remote/keycloak_config.dart
import 'package:flutter/foundation.dart';

/// Keycloak OIDC (browser PKCE) → API `/auth/oidc` exchange.
class KeycloakConfig {
  static String get url {
    const fromEnv = String.fromEnvironment('KEYCLOAK_URL');
    if (fromEnv.isNotEmpty) return fromEnv.replaceAll(RegExp(r'/$'), '');
    if (kIsWeb) return 'http://auth.localhost:8080';
    // Android emulator → host nginx
    return 'http://10.0.2.2:8080';
  }

  static const realm = String.fromEnvironment(
    'KEYCLOAK_REALM',
    defaultValue: 'edu-app',
  );

  static const clientId = String.fromEnvironment(
    'KEYCLOAK_CLIENT_ID',
    defaultValue: 'nihongo-mobile',
  );

  static String get authority => '$url/realms/$realm';

  static String get discoveryUrl =>
      '$authority/.well-known/openid-configuration';

  /// Flutter Android/iOS AppAuth redirect.
  static const redirectUrl = 'com.edu.nihongo_app:/oauth2redirect';
}

File: apps/nihongo-ios/NihongoEDU/Core/KeycloakConfig.swift
import Foundation

enum KeycloakConfig {
    static let url: String = {
        if let override = ProcessInfo.processInfo.environment["KEYCLOAK_URL"], !override.isEmpty {
            return override.hasSuffix("/") ? String(override.dropLast()) : override
        }
        #if targetEnvironment(simulator)
        return "http://localhost:8080"
        #else
        return "http://192.168.1.100:8080"
        #endif
    }()

    static let realm: String =
        ProcessInfo.processInfo.environment["KEYCLOAK_REALM"] ?? "edu-app"

    static let clientId: String =
        ProcessInfo.processInfo.environment["KEYCLOAK_CLIENT_ID"] ?? "nihongo-mobile"

    static var authority: String { "\(url)/realms/\(realm)" }

    static var authorizationEndpoint: String {
        "\(authority)/protocol/openid-connect/auth"
    }

    static var tokenEndpoint: String {
        "\(authority)/protocol/openid-connect/token"
    }

    /// Must match realm client redirect + CFBundleURLSchemes
    static let redirectURI = "com.edu.nihongo.ios:/oauth2redirect"
    static let callbackScheme = "com.edu.nihongo.ios"
}
